pipeline-examples/jobs-in-parallel/jobs_in_parallel.groovy

```diff
@@ -7,13 +7,14 @@ def branches = [:]
 //and jenkins will unite them into a single run of the job
 
 for (int i = 0; i < 4; i++) {
+  def _i = i //if we tried to use i below, it would equal 4 in each job execution.
```

Comment:
the prefixed underscore seems like it was an old syntax for designating a static variable? and it appears that it is no longer the recommended syntax. So I'm gonna force push this to `index` instead.

I don't think that there is specific documentation other then the stackoverflow that i linked in the PR for this, because as far as I can tell, this isn't so much specific to groovy, as it is with jenkins and the way it handles these variables, but there is no documentation to back that up ether.  So I think this will be as good as it gets.